x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutAutoFollowPatternAction.java

```diff
@@ -186,17 +208,53 @@ private static void markExistingIndicesAsAutoFollowedForNewPatterns(
             .stream()
             .filter(p -> previousPattern.getLeaderIndexPatterns().contains(p) == false)
             .collect(Collectors.toList());
-        markExistingIndicesAsAutoFollowed(newPatterns, leaderMetadata, followedIndexUUIDS);
+        markExistingIndicesAsAutoFollowed(newPatterns, leaderIndexExclusionPatterns, leaderMetadata, followedIndexUUIDS);
+    }
+
+    private static void warnIfFollowedIndicesExcludedWithNewPatterns(String autoFollowName,
+                                                                     List<String> newLeaderPatterns,
+                                                                     List<String> newLeaderIndexExclusionPatterns,
+                                                                     Metadata remoteMetadata,
+                                                                     Metadata localMetadata,
+                                                                     AutoFollowPattern previousPattern,
+                                                                     List<String> followedIndexUUIDS) {
+        final boolean hasNewExclusionPatterns = newLeaderIndexExclusionPatterns
+            .stream()
+            .anyMatch(p -> previousPattern.getLeaderIndexExclusionPatterns().contains(p) == false);
+
+        if (hasNewExclusionPatterns == false) {
+            return;
+        }
+
+        for (IndexMetadata localIndexMetadata : localMetadata) {
+            final Map<String, String> ccrMetadata = localIndexMetadata.getCustomData(Ccr.CCR_CUSTOM_METADATA_KEY);
+            if (ccrMetadata != null && followedIndexUUIDS.contains(ccrMetadata.get(Ccr.CCR_CUSTOM_METADATA_LEADER_INDEX_UUID_KEY))) {
+                final String leaderIndexName = ccrMetadata.get(Ccr.CCR_CUSTOM_METADATA_LEADER_INDEX_NAME_KEY);
+                IndexAbstraction indexAbstraction = remoteMetadata.getIndicesLookup().get(leaderIndexName);
+                final IndexMetadata leaderIndexMetadata = remoteMetadata.index(leaderIndexName);
+
+                if (AutoFollowPattern.match(newLeaderPatterns, newLeaderIndexExclusionPatterns, indexAbstraction) == false) {
+                    logger.warn("The follower index {} for leader index {} does not match against the updated auto follow " +
```

Comment:
I'm a bit torn about this - I wonder if we should make it more explicit to the user by returning an error. The same question applies when inclusion patterns are updated - a follower index might previously match a pattern but not anymore. Maybe we should prevent such update and require either a new pattern to be created or the follower indices to be unfollowed first?

Reply:
> In any case I think it's sensible to return an error in this scenario, I'll include it +1.

Maybe wait for @henningandersen's opinion first

Reply:
We discussed this offline and we agreed that instead of emitting a warning or failing the request when an Auto-Follow patter is updated and previously configured follower indices don't match anymore we'll keep the current behaviour. Instead we'll improve the current docs.